src/profile-logic/call-tree.js

```diff
@@ -492,11 +475,14 @@ function _getStackSelf(
     const callNodeIndex = sampleIndexToCallNodeIndex[sampleIndex];
     if (callNodeIndex !== null) {
       const weight = samples.weight ? samples.weight[sampleIndex] : 1;
-      callNodeSelf[callNodeIndex] += weight;
+      const leafBefore = callNodeLeaf[callNodeIndex];
+      const leafAfter = leafBefore + weight;
+      callNodeLeaf[callNodeIndex] = leafAfter;
+      rootTotalSummary += abs(leafAfter) - abs(leafBefore);
```

Comment:
Ah yes, I agree this deserves some justification.

This was an optimization for the performance of dragging a preview selection. Computing the rootTotalSummary was done as follows: sum over callNodeIndex of abs(callNodeSelf[callNodeIndex]). But when dragging a preview selection, many call nodes have zero leaf time. So it was faster to compute the sum per sample rather than per call node, because it avoids iterating over all the zero-time call nodes.

Actually, writing it out like this, it's possible that this is just a tradeoff; for profiles with fewer call nodes but lots more samples, maybe the old way to compute it is faster.

Before:
weights: [1, 2, 3], leaf: 0+1=1, 1+2=3, 3+3=6, abs: 6
weights: [1, 1, -2], leaf: 0+1=1, 1+1=2, 2+(-2)=0, abs: 0
weights: [1, 3, -2], leaf: 0+1=1, 1+3=4, 4+(-2)=2, abs: 2
weights: [-4, 5, -2], leaf: 0+(-4)=-4, -4+5=1, 1+(-2)=-1, abs: 1

After:
weights: [1, 2, 3], absleaf: 0 + (abs(0+1) - abs(0)) = 1, 1 + (abs(1+2) - abs(1)) = 3, 3 + (abs(3+3) - abs(3)) = 6
weights: [1, 1, -2], absleaf: 0 + (abs(0+1) - abs(0)) = 1, 1 + (abs(1+1) - abs(1)) = 2, 2 + (abs(2+(-2)) - abs(2)) = 0
weights: [1, 3, -2], absleaf: 0 + (abs(0+1) - abs(0)) = 1, 1 + (abs(1+3) - abs(1)) = 4, 4 + (abs(4+(-2)) - abs(4)) = 2
weights: [-4, 5, -2], absleaf: 0 + (abs(0+(-4)) - abs(0)) = 4, 4 + (abs(-4+5) - abs(-4)) = 1, 1 + (abs(1+(-2)) - abs(1)) = 1

Reply:
I'm going to revert this change. We can still make the change once the fast-invert work is done, it's quite orthogonal.

Reply:
> That said we don't seem to use it in the compare view.

What did you mean by this? We don't use the traced timing in the compare view (because we detect that sample weights are present), but we do use rootTotalSummary (from the sample weights) in the diff call tree.

Reply:
Ah I see, I totally missed that we weren't looping over the same table. And I was confused with callNodeIndex vs sampleIndex too. (not enough sleep)

> > That said we don't seem to use it in the compare view.
> 
> What did you mean by this? We don't use the traced timing in the compare view (because we detect that sample weights are present), but we do use rootTotalSummary (from the sample weights) in the diff call tree.

I just wasn't sure where we used it,  But I guess we use it to get the percentages in the call tree?

Reply:
Yes exactly, this is the value that the percentages are relative to.